pinot-spi/src/main/java/org/apache/pinot/spi/utils/EqualityUtils.java

```diff
@@ -156,9 +156,12 @@ public static boolean isEqualIgnoreOrder(@Nullable List left, @Nullable List rig
 
   public static boolean isEqualSet(@Nullable Set left, @Nullable Set right) {
     if (left != null && right != null) {
-      return isEqualIgnoreOrder(Arrays.asList(left.toArray()), Arrays.asList(right.toArray()));
+      if (left.size() != right.size()) {
+        return false;
+      }
+      return left.containsAll(right) && right.containsAll(left);
```

Comment:
When they are the same size, checking one side contains all of the other side should be enough

Reply:
Sounds good. I used this format since this is what's used in `isEqualMap`, but it is definitely redundant for sets.